### DISCUSSION TITLE
New firmware not working with Integration

### DISCUSSION THREAD

**Discussion by anothermort** [danielrivard/homeassistant-innova#194] [General]
I have a new unit fitted last week. The Innova TwoPointZero app no longer works with the newer units they have changed the setup process to be bluetooth 5.0. Also have been told the firmware has changed which I suspected is why the manual was so wrong.

New one makes more sense
https://www.innovaenergie.com/site/assets/files/2792/n420714b_manuale_installazione_rev_01_en-1.pdf

Newer app is now
https://play.google.com/store/apps/details?id=tech.solutiontech.innova

Could support for this be added to the integration or is their anything I can do to obtain info from my unit to create support for the newer firmware?

Happy to help if I can do anything.

Thanks

Adam

**Comment by anothermort** [comment]
Update from new app support team
"
Thanks for the kind feedback.

At the moment the APIs are not public, several firmware developments are still underway. Once completed there is the intention to publish them and also develop official plugins for HomeBridge, Alexa, etc...

I share your feedback with the development team so they know about your request!

Sorry for the inconvenience and thank you for the feedback.

Have a nice day, kind regards.

Innova IoT Support Team
"

**Reply by danielrivard** [reply]
That is an interesting information.

**Comment by danielrivard** [comment]
Well, this is unfortunate. I'll look at the new app and check if the calls it makes to the unit are similar.

Does it still connect to Wifi? I hope so.

If so is there some output from this URL: http://{{THE_IP_OF_YOUR_UNIT}}/api/v/1/status 
(maybe try with 2 or 1.1 instead of 1 as the version)

I probably won't have time to look into this in the short term, but I'll try to at least gather some information.

I wonder if they will update the firmware on the old units.

**Reply by anothermort** [reply]
Yes it does still connect to wifi just the setup is via bluetooth.

http://{{THE_IP_OF_YOUR_UNIT}}/api/v/1/status
tried a few versions 1,2,1.0,1.1, 2.0, 1_0 and 2_0
All came back with
This site can’t be reached
192.168.0.177 
refused to connect.

can ping it fine.

Not a problem with timing just thought I would offer to help. If I can run anything my end let me know.
Is there anything wireshark etc.. I can run to try to get more info?

I do now it was an Espressif device as I renamed it after it connect to my router.

They company I bought mine from only know about the software change because I told them. LOL

**Reply by danielrivard** [reply]
If you can do some wireshark and find some of the API calls it makes, that would help, but I doubt my current implementation will be compatible.
The [homeassistant](https://github.com/danielrivard/homeassistant-innova) part (this repo) is not really the problem, but the library I made for the [communication](https://github.com/danielrivard/innova-controls) would need a big rearchitecture. 
Or a new library that the integration could use to talk with the new firmware/api.

Also, looking at their new app, there seems to be some authentication needed that the previous one did not need. That would also complicate things.

**Reply by anothermort** [reply]
Will try wireshark in the next few days.
Have found an android app which will do the same for the app.

Hopefully something can come out of it eventually

Thanks

**Comment by danielrivard** [comment]
The new app is published by a different company than the previous one, I have a feeling that this is a complete rewrite and that will require a new design for my integration.

**Comment by anothermort** [comment]
If I can help to get some info.
Could do a fork of it maybe and see if that works. 
Happy to help.
